src/renderer/components/wallet/settings/WalletSettings.tsx

```diff
@@ -96,11 +97,21 @@ export const WalletSettings: React.FC<Props> = (props): JSX.Element => {
 
   const renderAddress = useCallback(
     (chain: Chain, { type: walletType, address: addressRD }: WalletAddress) => {
-      // Render ADD LEDGER button
       const renderAddLedger = (chain: Chain, loading: boolean) => (
-        <Styled.AddLedgerButton loading={loading} onClick={() => addLedgerAddress(chain)}>
-          <Styled.AddLedgerIcon /> {intl.formatMessage({ id: 'ledger.add.device' })}
-        </Styled.AddLedgerButton>
+        <Styled.AddLedgerContainer>
+          <Styled.AddLedgerButton loading={loading} onClick={() => addLedgerAddress(chain, walletIndex)}>
+            <Styled.AddLedgerIcon /> {intl.formatMessage({ id: 'ledger.add.device' })}
+          </Styled.AddLedgerButton>
+          {isBnbChain(chain) && (
+            <Styled.WalletIndexInput
+              value={walletIndex.toString()}
+              pattern="[0-9.]+"
```

Comment:
Remove `.` Maybe regex `\d` (digits only) will work, too?

Reply:
This one is maybe a bit more readable? But